lib/child_process.js

```diff
@@ -61,7 +62,9 @@ exports.fork = function(modulePath /*, args, options*/) {
 
   if (pos < arguments.length && arguments[pos] != null) {
     if (typeof arguments[pos] !== 'object') {
-      throw new TypeError('Incorrect value of args option');
+      throw new errors.TypeError('ERR_INVALID_OPT_VALUE',
+                                 ('arguments[' + pos + ']'),
+                                 arguments[pos]);
```

Comment:
This one feels super odd to me... will think about this a bit more.

Reply:
Shouldn't this be a `ERR_INVALID_ARG_TYPE` though? Judging from the code below I would say it's `options` that has the wrong type?